pkg/minikube/kubeconfig/config.go

```diff
@@ -81,8 +81,10 @@ func CacheSystemAdminEntries(systemEntriesConfigPath, clusterName string) error
 			break
 		}
 	}
-
 	userName := fmt.Sprintf("system:admin/%s", clusterName)
+	if isVersion3_6 {
+		userName = "system:admin/127-0-0-1:8443"
+	}
 	for k, v := range config.Users {
 		if v.Name == userName || v.Name == "system:admin/127-0-0-1:8443" {
```

Comment:
@praveenkumar @budhrg there bug in the code is here: 

    if v.Name == userName || v.Name == "system:admin/127-0-0-1:8443"

To start with the _v.Name == "system:admin/127-0-0-1:8443_ is generally not needed anymore with the introduction of the _isVersion3_6_ flag. If __isVersion3_6_ is true, both condition are actually identical. However, what is worse, it makes things fail independent of the OpenShift version. config.Users is iterating over a map which does not guarantee any order. Remember the 127-0-0-1:8443 is not gone form the config. It seems that on Linux the iteration order is different than on OS X. Removing _|| v.Name == "system:admin/127-0-0-1:8443"_ should do the trick.